qiskit/circuit/library/data_preparation/state_preparation.py

```diff
@@ -414,3 +439,131 @@ def _multiplex(self, target_gate, list_of_angles, last_cnot=True):
             circuit.append(CXGate(), [msb, lsb])
 
         return circuit
+
+
+class Generalized_Uniform_Superposition_Gate(Gate):
+    """
+    Generates a generalized uniform superposition state, $\frac{1}{\sqrt{M}} \sum_{j=0}^{M-1}  \ket{j} $ (where 1< M <= 2^n), 
+    using n qubits, following the Shukla-Vedula algorithm [SV24].
+
+    Note: The Shukla-Vedula algorithm [SV24] provides an efficient approach for creation of a generalized uniform superposition 
+    state of the form, $\frac{1}{\sqrt{M}} \sum_{j=0}^{M-1}  \ket{j} $. It needs only $O(\log_2 (M))$ qubits and $O(\log_2 (M))$ 
+    gates, hence providing an exponential improvement, in terms of reduced resources and complexity, compared to alternative methods
+    in existing literature.
+    
+    Args:
+        M (int): 
+            A positive integer M (> 1) representing the number of computational basis states with an amplitude of 1/sqrt(M) 
+            in the uniform superposition state ($\frac{1}{\sqrt{M}} \sum_{j=0}^{M-1}  \ket{j} $). Note that the remaining 
+            (2^n - M) computational basis states have zero amplitudes. Here M need not be an integer power of 2.
+            
+        num_qubits (int): 
+            A positive integer representing the number of qubits used.
+
+    Returns:
+        cirq.Circuit: A quantum circuit that creates the uniform superposition state: $\frac{1}{\sqrt{M}} \sum_{j=0}^{M-1}  \ket{j} $. 
```

Comment:
I think its more normal to have the Args and descriptions thereof just in the init doc string, like StatePreparation does. That will hopefully also take care of the return type here!!!

Reply:
Thanks for the suggestion, Steve. I just pushed your recommendation.

Reply:
I should have looked more carefully - you have `M` as a public instance variable which is maybe why you have it in Args there. I am more used to seeing the public instance vars done under Attributes there. I guess you want that as a public var? I am noting that num_qubits is a private var though. StatePreparation they are all marked private i.e. a leading underscore in the name.